### PR TITLE
feat(inventory-plugin): filter droplets by tag during building inventory

### PR DESCRIPTION
**Pull request by krecik**
##### SUMMARY
Filter droplets by tag name. Useful to use, to ex. filter over one tag and later split inventory by other tags.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
Inventory plugin

##### ADDITIONAL INFORMATION
n/a

### CODE REVIEW

**Comment by mamercad** [comment]
Hello! Thanks for this contribution. I'm going over the [API docs](https://developers.digitalocean.com/documentation/v2/#droplets), and I'm reading this:

> To list Droplets by a tag, send a GET request to /v2/droplets?tag_name=$TAG_NAME.
> The response will match that of regular droplet listing request but will be filtered to only include the tagged Droplets.

In the example that you're updating, you've written `filter_by_tag: 'project-name'`, which is a little bit confusing (DigitalOcean *does* have Projects, and they're different than tags). Perhaps update this to be `filter_by_tag: 'my-tag'`?

Also, could you submit a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) with this please?

**Review by mamercad — CHANGES_REQUESTED** [review]
See my comments in the conversation please.

**Comment by krecik** [comment]
Added changelog fragments file. Also changed example as suggested @mamercad

**Comment by krecik** [comment]
Added unit test for filter_by_tag option.

File: changelogs/fragments/77-filter-droplets-by-tag.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - digitalocean - filter droplets by tag in dynamic inventory plugin
```
Comment (mamercad) [inline changelogs/fragments/77-filter-droplets-by-tag.yml:2]: Mind rewriting this as `- digitalocean - filter droplets by tag in dynamic inventory plugin (https://github.com/ansible-collections/community.digitalocean/pull/77)` please?

Comment (krecik) [reply changelogs/fragments/77-filter-droplets-by-tag.yml:2]: OK

**Comment by mamercad** [comment]
Closing since #96 was merged.

**Pull request closed without merging**